File: packages/template-hello-world/app/main-view-model.js
var observable = require("data/observable");

var counter = 42;

var mainViewModel = new observable.Observable();
mainViewModel.set("message", counter + " taps left");
mainViewModel.tapAction = function () {
   counter--;
   if (counter <= 0) {
     mainViewModel.set("message", "Hoorraaay! You unlocked the NativeScript clicker achievement!");
   }
   else {
     mainViewModel.set("message", counter + " taps left");
   }
};
exports.mainViewModel = mainViewModel;

File: packages/template-hello-world-kt/app/main-view-model.kt
fun createViewModel() = Observable().apply {
    // initialize observable's properties:
    
    val initialCounterValue = 42

    set("counter", initialCounterValue)
    set("message", getMessage(initialCounterValue))

    set("onTap") { 
        val currentCounterValue = get<Int>("counter")
        val newCounterValue = currentCounterValue - 1

        set("counter", newCounterValue)
        set("message", getMessage(newCounterValue))
    }
}

private fun getMessage(counter: Int) = 
    if (counter <= 0) "Hoorraaay! You unlocked the NativeScript clicker achievement!" 
    else "${counter} taps left"